models/users.js

```diff
@@ -42,9 +42,17 @@ const addOrUpdate = async (userData, userId = null) => {
       const isNewUser = !user.data();
       // user exists update user
       if (user.data()) {
+        // remove id field if exist in fetched data or profileDiff
+        const currentData = user.data();
+        if ("id" in currentData) {
+          delete currentData.id;
+        }
+        if ("id" in userData) {
+          delete userData.id;
+        }
         await userModel.doc(userId).set(
           {
-            ...user.data(),
+            ...currentData,
```

Comment:
Can you create an issue for removing this current data as it will not make any difference as we merge our data?

Reply:
Create #2261 for the same. Please check